### PR TITLE
Using a custom dialer for cluster hosts

### PR DESCRIPTION
**Pull request by galal-hussein**

### CODE REVIEW

File: cmd/remove.go [hunk 1]
```diff
@@ -72,5 +73,5 @@ func clusterRemoveFromCli(ctx *cli.Context) error {
 	if err != nil {
 		return fmt.Errorf("Failed to resolve cluster file: %v", err)
 	}
-	return ClusterRemove(clusterFile)
+	return ClusterRemove(clusterFile, nil)
```
Comment (alena1108) [inline cmd/remove.go]: Just to confirm that is when the code is called from cli context, the connection should be done using old way - ssh. And the customDialer will be used only when called from the kontainer-engine

File: cmd/up.go [hunk 2]
```diff
@@ -27,10 +28,10 @@ func UpCommand() cli.Command {
 	}
 }
 
-func ClusterUp(clusterFile string) (string, string, string, string, error) {
+func ClusterUp(clusterFile string, customDialer hosts.CustomDialer) (string, string, string, string, error) {
```
Comment (alena1108) [inline cmd/up.go]: doesn't look like all the values this method return, being used? why not just return an error then?

Comment (galal-hussein) [reply cmd/up.go]: these are the values that kontainer-engine requested like `APIURL`,`cacrt`,`cakey`, etc

Comment (alena1108) [reply cmd/up.go]: Unless they are used when called from kontainer-engine?

**Pull request merged**